src/Components/DataHistoryExplorer/Internal/TimeSeriesTwinCallout/TimeSeriesTwinCallout.tsx

```diff
@@ -0,0 +1,235 @@
+        onPrimaryActionClick,
+        styles
+    } = props;
+
+    // state
+    const [
+        timeSeriesTwinToEdit,
+        setTimeSeriesTwinToEdit
+    ] = useState<IDataHistoryTimeSeriesTwin>(
+        deepCopy(timeSeriesTwin || defaultTimeSeriesTwin)
+    );
+    const isLabelAutoPopulated = useRef(false);
+    const isAddDisabled = !(
+        timeSeriesTwinToEdit.twinId && timeSeriesTwinToEdit.twinPropertyName
+    );
+
+    // hooks
+    const { t } = useTranslation();
+
+    // callbacks
+    const handleTwinIdChange = useCallback((selectedTwinId: string) => {
+        setTimeSeriesTwinToEdit(
+            produce((draft) => {
+                draft.twinId = selectedTwinId;
+                draft.twinPropertyName = '';
+                draft.twinPropertyType = null;
+                if (isLabelAutoPopulated.current) {
+                    draft.label = '';
+                }
+            })
+        );
+    }, []);
+    const handleTwinPropertyChange = useCallback(
+        (newPropertyExpression: PropertyExpression) => {
+            setTimeSeriesTwinToEdit(
+                produce((draft) => {
+                    draft.twinPropertyName =
+                        newPropertyExpression.property.localPath; // not assigning property 'name' but 'localPath' instead in case the property is a nested one
+                    draft.twinPropertyType =
+                        newPropertyExpression.property.propertyType;
+
+                    if (!draft.label) {
+                        draft.label = `${draft.twinId} ${newPropertyExpression.property.localPath}`; // auto populate the label with selected twin and its property name if it is empty
+                        isLabelAutoPopulated.current = true;
+                    }
+                })
+            );
+        },
+        []
+    );
+    const handleTwinPropertyCastingChange = useCallback(
+        (_event, checked: boolean) => {
+            setTimeSeriesTwinToEdit(
+                produce((draft) => {
+                    draft.chartProps.isTwinPropertyTypeCastedToNumber = checked;
+                })
+            );
+        },
+        []
+    );
+    const handleLabelChange = useCallback((_event, label: string) => {
+        setTimeSeriesTwinToEdit(
+            produce((draft) => {
+                draft.label = label;
+                isLabelAutoPopulated.current = false;
+            })
+        );
+    }, []);
+
+    // styles
+    const classNames = getClassNames(styles, {
+        theme: useTheme()
+    });
+
+    return (
+        <DataHistoryExplorerContext.Consumer>
+            {({ adapter }) => (
+                <div className={classNames.root}>
+                    <Callout
+                        target={`#${target}`}
+                        styles={classNames.subComponentStyles.callout}
+                        onDismiss={onDismiss}
+                        directionalHint={DirectionalHint.bottomCenter}
+                        isBeakVisible={false}
```

Comment:
why no beak?

Reply:
as far as I know we have not been using beak in callouts in the builder components, no? Not sure though, I can confirm it with Azima.

Reply:
added beak and positioned the callout to right center